src/codeu/chat/util/Tokenizer.java

```diff
@@ -0,0 +1,76 @@
+package codeu.chat.util;
+
+import java.io.IOException;
+
+public final class Tokenizer {
+   private StringBuilder token;
+   private String source;
```

Comment:
also, please move final source arg to the top, grouping final and non-final ones together